packages/core/src/api/exporters/copyExtension.ts

```diff
@@ -17,46 +52,79 @@ export const createCopyToClipboardExtension = <
   Extension.create<{ editor: BlockNoteEditor<BSchema, I, S> }, undefined>({
     name: "copyToClipboard",
     addProseMirrorPlugins() {
-      const tiptap = this.editor;
-      const schema = this.editor.schema;
       return [
         new Plugin({
           props: {
             handleDOMEvents: {
-              copy(_view, event) {
+              copy(view, event) {
                 // Stops the default browser copy behaviour.
                 event.preventDefault();
                 event.clipboardData!.clearData();
 
-                const selectedFragment =
-                  tiptap.state.selection.content().content;
-
-                const internalHTMLSerializer = createInternalHTMLSerializer(
-                  schema,
-                  editor
-                );
-                const internalHTML =
-                  internalHTMLSerializer.serializeProseMirrorFragment(
-                    selectedFragment
-                  );
-
-                const externalHTMLExporter = createExternalHTMLExporter(
-                  schema,
-                  editor
-                );
-                const externalHTML =
-                  externalHTMLExporter.exportProseMirrorFragment(
-                    selectedFragment
+                // Checks if a `blockContent` node is being copied and expands
+                // the selection to the parent `blockContainer` node.
+                if (
+                  "node" in view.state.selection &&
+                  (view.state.selection.node as Node).type.spec.group ===
+                    "blockContent"
+                ) {
+                  view.dispatch(
+                    view.state.tr.setSelection(
+                      new NodeSelection(
+                        view.state.doc.resolve(view.state.selection.from - 1)
+                      )
+                    )
                   );
+                }
 
-                const plainText = cleanHTMLToMarkdown(externalHTML);
+                const { internalHTML, externalHTML, plainText } =
+                  selectedFragmentToHTML(view, editor);
 
                 // TODO: Writing to other MIME types not working in Safari for
                 //  some reason.
                 event.clipboardData!.setData("blocknote/html", internalHTML);
                 event.clipboardData!.setData("text/html", externalHTML);
                 event.clipboardData!.setData("text/plain", plainText);
 
+                // Prevent default PM handler to be called
+                return true;
+              },
+              dragstart(view, event) {
```

Comment:
similar to above comment; it's unclear for me why we now also need the dragstart event. What was fixed by this? Can we add a comment to explain for which scenario we need the dragstart event?